modules/index/src/test/java/org/apache/ignite/internal/index/IndexManagementUtilsTest.java

```diff
@@ -66,4 +86,36 @@ void testExtractIndexIdFromPartitionBuildIndexKey() {
         assertEquals(1, extractIndexIdFromPartitionBuildIndexKey("indexBuild.partition.1.2"));
         assertEquals(7, extractIndexIdFromPartitionBuildIndexKey("indexBuild.partition.7.9"));
     }
+
+    @Test
+    void testIsPrimaryReplicaTrue() {
+        TablePartitionId replicaGroupId = new TablePartitionId(1, 0);
+
+        HybridTimestamp startTime = clock.now();
+        long dayInMillis = TimeUnit.DAYS.toMillis(1);
+
+        ReplicaMeta replicaMeta = newPrimaryReplicaMeta(LOCAL_NODE, replicaGroupId, startTime, startTime.addPhysicalTime(dayInMillis));
+
+        assertTrue(isPrimaryReplica(replicaMeta, LOCAL_NODE, clock.now()));
+    }
+
+    @Test
+    void testIsPrimaryReplicaFalse() {
```

Comment:
This test seems to actually have 2 test cases inside, I would split it into 2, but it's up to you

Reply:
I'll leave it as it is.